### PR TITLE
summarystats improvements

### PR DESCRIPTION
**Pull request by itsdebartha**
This commit addresses issue #693 
- Add standard deviation to summarystats
- Update documentation
- Add tests

### CODE REVIEW

**Comment by itsdebartha** [comment]
`sem` returns `ArgumentError` but expects `MethodError` in CI. However, in my machine, `sem` returns `MethodError`, and passes the tests.

**Comment by ParadaCarleton** [comment]
The tests are failing on Julia nightly, which is fine (it usually means there's a bug in the nightly build, which is pre-alpha, rather than a bug in our code).

**Comment by itsdebartha** [comment]
Yeah, seems like `MethodError` has changed to `ArgumentError` for some of the results...

File: src/scalarstats.jl [hunk 1]
```diff
@@ -871,14 +872,16 @@ end
     summarystats(a)
 
 Compute summary statistics for a real-valued array `a`. Returns a
-`SummaryStats` object containing the mean, minimum, 25th percentile,
-median, 75th percentile, and maxmimum.
+`SummaryStats` object containing the number of observations,
+number of missing observations, standard deviation, mean, minimum,
+25th percentile, median, 75th percentile, and maxmimum.
```
Comment (nalimilan) [inline src/scalarstats.jl:877]: While we're at it:
```suggestion
25th percentile, median, 75th percentile, and maximum.
```

File: test/scalarstats.jl [hunk 2]
```diff
@@ -332,30 +332,39 @@ s = summarystats(1:5)
 @test isa(s, StatsBase.SummaryStats)
 @test s.min == 1.0
 @test s.max == 5.0
+@test s.nobs   == 5
+@test s.nmiss  == 0
 @test s.mean   ≈ 3.0
 @test s.median ≈ 3.0
 @test s.q25    ≈ 2.0
 @test s.q75    ≈ 4.0
+@test s.sd     ≈ 1.5 atol=1e-1
```
Comment (nalimilan) [inline test/scalarstats.jl:341]: Use a lower tolerance? I guess you should be able to keep the default tolerance if you copy the actual value.

**Review by nalimilan — COMMENTED** [review]
Thanks! Sorry for the delay.

File: src/scalarstats.jl (hunk 1 above)
Comment (itsdebartha) [reply src/scalarstats.jl]: Thanks for pointing out this mistake...

File: test/scalarstats.jl (hunk 2 above)
Comment (itsdebartha) [reply test/scalarstats.jl]: Yeah, it's updated now...

File: src/deviation.jl [hunk 3]
```diff
@@ -12,7 +12,7 @@ function counteq(a::AbstractArray, b::AbstractArray)
     n = length(a)
     length(b) == n || throw(DimensionMismatch("Inconsistent lengths."))
     c = 0
-    for i = 1:n
+    for i in eachindex(a, b)
```
Comment (nalimilan) [inline src/deviation.jl:15]: Thanks for doing this (it's been long due!), but can you move this to another PR?

Comment (itsdebartha) [reply src/deviation.jl]: Yup, I have done it...

**Comment by itsdebartha** [comment]
@nalimilan I have made the corrections. Hopefully, it is ok now...

File: src/scalarstats.jl [hunk 4]
```diff
@@ -871,14 +872,16 @@ end
...
+`SummaryStats` object containing the number of observations,
+number of missing observations, standard deviation, mean, minimum,
+25th percentile, median, 75th percentile, and maximum.
 """
 function summarystats(a::AbstractArray{T}) where T<:Union{Real,Missing}
     # `mean` doesn't fail on empty input but rather returns `NaN`, so we can use the
     # return type to populate the `SummaryStats` structure.
     s = T >: Missing ? collect(skipmissing(a)) : a
     m = mean(s)
+    stdev = std(s)
```
Comment (nalimilan) [inline src/scalarstats.jl:884]: Actually this should be a bit more efficient:
```suggestion
    stdev = std(s, mean=m)
```
